src/PostgREST/ApiRequest.hs

```diff
@@ -271,3 +277,85 @@ ensureUniform arr =
   if (V.length objs == V.length arr) && areKeysUniform
     then Just (UniformObjects objs)
     else Nothing
+
+{-|
+  Test whether a proxy uri is malformed or not.
+  A valid proxy uri should be an absolute uri without query and user info,
+  only http(s) schemes are valid, port number range is 1-65535.
+
+  For example
+  http://postgrest.com/openapi.json
+  https://postgrest.com:8080/openapi.json
+-}
+isMalformedProxyUri :: Maybe BS.ByteString -> Bool
+isMalformedProxyUri Nothing =  False
+isMalformedProxyUri (Just uri)
```

Comment:
considering the fact that this affects only the json output for the schema and that most of the time (if not all) the person deploying postgrest will also be the one in control of the proxy, what's the downside of just trusting the uri is valid ... treat it as a string?